src/App.tsx

```diff
@@ -3,33 +3,49 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handelChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ query: event.target.value });
+  };
 
   render() {
+    const { query } = this.state;
+
+    const visibleMovies = moviesFromServer.filter(
+      movie => movie.title.toLowerCase().includes(query.toLowerCase())
+      || movie.description.toLowerCase().includes(query.toLowerCase()),
+    );
```

Comment:
Мб это надо будет вынести в отдельный метод

Reply:
create getVisibleMovies method
```suggestion
    const visibleMovies = this.getVisibleMovies(moviesFromServer);
```